File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> sorted_arr = arr;

        sort(sorted_arr.begin() , sorted_arr.end());
        unordered_map<int , int>rank_map;
        int rank = 1;

        for(int num : sorted_arr){
            if(rank_map.find(num) == rank_map.end()){
                rank_map[num] = rank;
                rank++;
            }
        }

        for (int i = 0 ; i < arr.size(); i++){
            arr[i] = rank_map[arr[i]];
        }

        return arr;
    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        sorted_unique = sorted((set(arr)))

        rank_map = {}

        for rank , num in enumerate(sorted_unique , 1):
            rank_map[num] = rank

        return [rank_map[num] for num in arr]